### PR TITLE
Update dashboard build to use webpack v5

### LINKED ISSUE #5945: Update webpack to v5
The project still uses webpack v4; but we probably want to be on v5 to pick up more recent dependencies (and security fixes).

See https://webpack.js.org/migrate/5/

### PR DESCRIPTION
**Pull request by dadjeibaah**
This change upgrades the current webpack version used to build the
dashboard to version 5. This upgrade required an update to some
deprecated js build libraries (i.e. `@babel-eslint` ->
`@babel/eslint-parser` and `eslint-loader` -> `eslint-webpack-plugin`)

Upgrading these packaged also added some new linting rules that. Some of
the files have been updated to fix them while some other linting rules
were disabled and will be fixed in a follow up PR.

This change also includes updates the the dashboard `Dockerfile` to
update the node and yarn versions used to build the web assets.

Fixes #5945

Signed-off-by: Dennis Adjei-Baah <dennis@buoyant.io>

### CODE REVIEW

File: bin/web [hunk 1]
```diff
@@ -55,7 +55,7 @@ dev() {
     esac
   done
 
-  cd "$ROOT"/web/app && yarn webpack-dev-server --port "$DEV_PORT" &
+  cd "$ROOT"/web/app && yarn webpack serve --port "$DEV_PORT" &
```
Comment (Pothulapati) [inline bin/web:58]: Does this support live-reloading that the dev server provides?

**Review by Pothulapati — APPROVED** [review]
LGTM

**Review by adleong — APPROVED** [review]

**Review by alpeb — APPROVED** [review]
LGTM modulo nit above :+1:

File: bin/web (hunk 1 above)
Comment (dadjeibaah) [reply bin/web:58]: Yes, live reloading should be on by default if webpack is started with `NODE_ENV` set to `development`.

**Pull request merged**